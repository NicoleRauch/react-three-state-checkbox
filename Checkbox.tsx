import * as React from 'react';
import { CheckboxProps } from './index';

export default class Checkbox extends React.Component<CheckboxProps, any> {
  static defaultProps: Partial<CheckboxProps> = {
    indeterminate: false,
    className: '',
    onChange: () => {}     // tslint:disable-line no-empty
  };

  render() {
    const { className, checked, indeterminate, onChange } = this.props;
    return (
      <input
        type="checkbox"
        className={className}
        ref={input => {
          if (input) {
            input.checked = checked;
            input.indeterminate = indeterminate as boolean;
          }
        }}
        onChange={onChange}
      />
    );
  }
}
