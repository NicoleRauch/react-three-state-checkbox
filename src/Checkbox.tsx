import * as React from 'react';
import { CheckboxProps } from '../index';

export default class Checkbox extends React.Component<CheckboxProps, any> {
  /* istanbul ignore next: callback */
  static defaultProps: Partial<CheckboxProps> = {
    indeterminate: false,
    className: '',
    onChange: () => {}     // tslint:disable-line no-empty
  };

  render() {
    const { className, checked, indeterminate, onChange } = this.props;
    return (
      <input
        type="checkbox"
        className={className}
        ref={input => {
          /* istanbul ignore else */
          if (input) {
            input.checked = checked;
            input.indeterminate = indeterminate as boolean;
          }
        }}
        onChange={onChange}
      />
    );
  }
}
