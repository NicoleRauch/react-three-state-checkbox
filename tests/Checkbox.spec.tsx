import * as React from 'react';
import { configure, shallow, mount } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

import Checkbox from '../src/Checkbox';

describe('<Checkbox />', () => {
  it('renders checkbox', () => {
    const wrapper = shallow(<Checkbox checked />);
    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('renders checkbox with the right default props', () => {
    const wrapper = mount(<Checkbox checked={false} />);
    const { checked, indeterminate, className } = wrapper.props();
    expect(checked).toEqual(false);
    expect(indeterminate).toEqual(false);
    expect(className).toEqual('');
  });

  it('renders checkbox with all props defined', () => {
    const wrapper = mount(<Checkbox checked indeterminate className="test-class" />);
    const { checked, indeterminate, className } = wrapper.props();
    expect(checked).toEqual(true);
    expect(indeterminate).toEqual(true);
    expect(className).toEqual('test-class');
  });
});
