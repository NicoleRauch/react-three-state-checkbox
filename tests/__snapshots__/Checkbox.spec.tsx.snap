// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders checkbox 1`] = `
<Checkbox
  checked={true}
  className=""
  indeterminate={false}
  onChange={[Function]}
>
  <input
    className=""
    onChange={[Function]}
    type="checkbox"
  />
</Checkbox>
`;

exports[`<Checkbox /> renders checkbox with all props defined 1`] = `
<Checkbox
  checked={true}
  className="test-class"
  indeterminate={true}
  onChange={[Function]}
>
  <input
    className="test-class"
    onChange={[Function]}
    type="checkbox"
  />
</Checkbox>
`;

exports[`<Checkbox /> renders checkbox with the right default props 1`] = `
<Checkbox
  checked={false}
  className=""
  indeterminate={false}
  onChange={[Function]}
>
  <input
    className=""
    onChange={[Function]}
    type="checkbox"
  />
</Checkbox>
`;
